### PR TITLE
Update XGBoost and LightGBM examples

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Update the XGboost and LightGBM examples to use automatic logging.

## How is this patch tested?

`pytest tests/examples/test_examples.py`

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: examples/lightgbm/train.py [hunk 1]
```diff
@@ -29,31 +29,32 @@ def main():
...
-
-    # log parameters, metrics, and model
-    with mlflow.start_run() as run:
-        mlflow.log_params(params)
+    train_set = lgb.Dataset(X_train, label=y_train)
+
+    # enable auto logging
+    mlflow.lightgbm.autolog()
+
+    with mlflow.start_run():
```
Comment (smurching) [inline examples/lightgbm/train.py]: If we use autologging, we shouldn't need a `mlflow.start_run` anymore, right?

File: examples/xgboost/train.py [hunk 2]
```diff
@@ -32,29 +32,30 @@ def main():
...
-    loss = log_loss(y_test, y_proba)
-    acc = accuracy_score(y_test, y_pred)
-
-    # log parameters, metrics, and model
-    with mlflow.start_run() as run:
-        mlflow.log_params(params)
+    # enable auto logging
+    mlflow.xgboost.autolog()
+
+    with mlflow.start_run():
```
Comment (smurching) [inline examples/xgboost/train.py]: Same here, wondering if we can remove the `mlflow.start_run`

**Review by smurching — COMMENTED** [review]
Just small comments, thanks @harupy!

File: examples/lightgbm/train.py (hunk 1 above)
Comment (harupy) [reply examples/lightgbm/train.py]: If we didn't use `mlflow.start_run` here, `log_metrics` after `lgb.train` would create a new run and record metrics in it.

**Comment by harupy** [comment]
`test_examples.py` in python 2.7 fails due to the following error:

https://travis-ci.org/mlflow/mlflow/jobs/645884094#L2468

```
_tkinter.TclError: no display name and no $DISPLAY environment variable
```

We can avoid this error by setting Matplotlib to use non-interactive backend with:
```python
import matplotlib
matplotlib.use('Agg')
```

However, I'm wondering if it's ok to add this code to the example scripts because users might think this code is required to use auto logging.

File: examples/lightgbm/train.py [hunk 3]
```diff
@@ -4,18 +4,21 @@
...
 
 def parse_args():
     parser = argparse.ArgumentParser(description='LightGBM example')
     parser.add_argument('--colsample-bytree', type=float, default=1.0,
                         help='subsample ratio of columns when constructing each tree (default: 1.0)')
     parser.add_argument('--subsample', type=float, default=1.0,
                         help='subsample ratio of the training instances (default: 1.0)')
-    parser.add_argument('--conda-env',  type=str, default=None,
+    parser.add_argument('--conda-env', type=str, default=None,
                         help='the path to a conda environment yaml file (default: None)')
```
Comment (smurching) [inline examples/lightgbm/train.py]: Let's remove this argument now that it's no longer used? Same in the xgboost example

**Review by smurching — APPROVED** [review]
Sorry for the delay @harupy, had one last observation, otherwise LGTM

**Comment by harupy** [comment]
@smurching 
Fixed the `fbprophet` issue and all the tests passed.

**Comment by harupy** [comment]
@smurching Thank you always!

**Pull request merged**